dot/parachain/backing/candidate_backing.go

```diff
@@ -5,124 +5,723 @@ package backing
+	ParentHash common.Hash
 }
 
 // ActiveLeavesUpdate is a messages from overseer
 type ActiveLeavesUpdate struct {
 	// TODO: Complete this struct #3503
 }
 
-// GetBackedCandidates is a message received from overseer that requests a set of backable
+// GetBackedCandidatesMessage is a message received from overseer that requests a set of backable
 // candidates that could be backed in a child of the given relay-parent.
-type GetBackedCandidates []struct {
+type GetBackedCandidatesMessage []struct {
 	CandidateHash        parachaintypes.CandidateHash
 	CandidateRelayParent common.Hash
 }
 
-// CanSecond is a request made to the candidate backing subsystem to determine whether it is permissible
+// CanSecondMessage is a request made to the candidate backing subsystem to determine whether it is permissible
 // to second a given candidate.
 // The rule for seconding candidates is: Collations must either be built on top of the root of a fragment tree
 // or have a parent node that represents the backed candidate.
-type CanSecond struct {
+type CanSecondMessage struct {
 	CandidateParaID      parachaintypes.ParaID
 	CandidateRelayParent common.Hash
 	CandidateHash        parachaintypes.CandidateHash
 	ParentHeadDataHash   common.Hash
 }
 
-// Second is a message received from overseer. Candidate Backing subsystem should second the given
+// SecondMessage is a message received from overseer. Candidate Backing subsystem should second the given
 // candidate in the context of the given relay parent. This candidate must be validated.
-type Second struct {
+type SecondMessage struct {
 	RelayParent             common.Hash
 	CandidateReceipt        parachaintypes.CandidateReceipt
 	PersistedValidationData parachaintypes.PersistedValidationData
 	PoV                     parachaintypes.PoV
 }
 
-// Statement represents a validator's assessment of a specific candidate. If there are disagreements
+// StatementMessage represents a validator's assessment of a specific candidate. If there are disagreements
 // regarding the validity of this assessment, they should be addressed through the Disputes Subsystem,
 // with the actual escalation deferred until the approval voting stage to ensure its availability.
 // Meanwhile, agreements are straightforwardly counted until a quorum is achieved.
-type Statement struct {
+type StatementMessage struct {
 	RelayParent         common.Hash
 	SignedFullStatement SignedFullStatementWithPVD
 }
 
+// SignedFullStatementWithPVD represents a signed full statement along with associated Persisted Validation Data (PVD).
+type SignedFullStatementWithPVD struct {
+	SignedFullStatement     parachaintypes.UncheckedSignedFullStatement
+	PersistedValidationData *parachaintypes.PersistedValidationData
+}
+
+// New creates a new CandidateBacking instance and initialises it with the provided overseer channel.
 func New(overseerChan chan<- any) *CandidateBacking {
 	return &CandidateBacking{
 		SubSystemToOverseer: overseerChan,
 	}
```

Comment:
Should not this function initialize the `perRelayParent` and `perCandidate` map?

```suggestion
	return &CandidateBacking{
		SubSystemToOverseer: overseerChan,
		perRelayParent: map[common.Hash]perRelayParentState{},
		perCandidate: map[common.Hash]perCandidateState{},
	}
```